artifacts/artifact_funcs.jl

```diff
@@ -59,3 +60,20 @@ function pr_obs_data_path()
     )
     return AW.get_data_folder(pr_obs_data)
 end
+
+"""
+    artifact_data(datapath_full, name; datapath_trunc, date0, time_start, time_end, comms_ctx)
```

Comment:
Maybe we should add docstrings for both methods of this function and explain what they do differently (i.e. one truncates the dataset and the other just returns the input dataset)